### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Olekk17**
[DEMO LINK](https://olekk17.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,143 @@
...
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const isTitleEmpty = title.length === 0;
+    const isUserNotChosen = selectedUser === null;
+
+    setIsTitleError(isTitleEmpty);
+    setIsUserError(isUserNotChosen);
+
+    if (!isUserNotChosen && !isTitleEmpty) {
```
Comment (VenomMB) [inline src/App.tsx:36]: It better move in a separate variable.

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,143 @@
...
+      setTodos(current => ([
+        ...current,
+        {
+          id: Math.max(...current.map(todo => todo.id)) + 1,
```
Comment (VenomMB) [inline src/App.tsx:40]: The same.

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,143 @@
...
+      setSelectedUser(null);
+      setIsUserError(false);
+      setIsTitleError(false);
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setIsTitleError(value.length === 0);
```
Comment (VenomMB) [inline src/App.tsx:58]:
```suggestion
    setIsTitleError(!value.length);
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,143 @@
...
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isUserError, setIsUserError] = useState(false);
+  const [isTitleError, setIsTitleError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const isTitleEmpty = title.length === 0;
+    const isUserNotChosen = selectedUser === null;
```
Comment (VenomMB) [inline src/App.tsx:31]:
```suggestion
    const isTitleEmpty = !title.length;
    const isUserNotChosen = !selectedUser;
```

**Review by Vlad-Lischenovskiy — APPROVED** [review]
Great Job!!!

**Review by VenomMB — APPROVED** [review]
GJ! Pay attention to comments.

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,146 @@
...
+              onChange={handleInputChange}
+              placeholder="Enter new title"
+            />
+          </label>
+          {isTitleError
+          && (
+            <span className="error">
+              Please enter a title
+            </span>
+          )}
```
[the remark this answers is not in the document]
Comment (SerhiiKirik) [reply src/App.tsx:100]: In all cases

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,146 @@
...
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isUserError, setIsUserError] = useState(false);
+  const [isTitleError, setIsTitleError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const isTitleEmpty = !title.length;
```
Comment (SerhiiKirik) [inline src/App.tsx:32]:
```suggestion
    event.preventDefault();
    
    const isTitleEmpty = !title.length;
```

**Review by SerhiiKirik — APPROVED** [review]
Well done 🔥 
Your code is getting better every time, keep it up 🚀 
Let's fix bugs 🧑‍💻 
image

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Sync 🌍

File: src/App.tsx [hunk 7]
```diff
@@ -1,61 +1,154 @@
...
+              onChange={handleSelectChange}
+            >
+              <option
+                value="0"
+                disabled
+              >
+                Choose a user
+              </option>
+              {usersFromServer.map((user) => (
+
```
Comment (SerhiiKirik) [inline src/App.tsx:124]:
```suggestion
              </option>
              
              {usersFromServer.map((user) => (
```

File: src/App.tsx [hunk 8]
```diff
@@ -1,61 +1,154 @@
...
+                >
+                  {user.name}
+                </option>
+              ))}
+            </select>
+          </label>
 
-          <span className="error">Please choose a user</span>
+          {isUserError
+          && (
```
Comment (SerhiiKirik) [inline src/App.tsx:136]: in all cases
```suggestion
          {isUserError && (
```

File: src/App.tsx [hunk 9]
```diff
@@ -1,61 +1,154 @@
...
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isTitleEmpty = !title.trim().length;
+    const isUserNotChosen = !selectedUser;
+    const isBothFieldsNotEmpty = !isUserNotChosen && !isTitleEmpty;
+
+    setIsTitleError(isTitleEmpty);
+    setIsUserError(isUserNotChosen);
```
Comment (SerhiiKirik) [inline src/App.tsx:38]: if you found errors - return from function

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Keep it up 🔥 
Let's fix this bug 🧑‍💻 
image

**Review by SerhiiKirik — APPROVED** [review]
LGTM 🔥

**Pull request closed without merging**